src/ProgramVariable.cpp

```diff
@@ -14,7 +14,7 @@ ProgramVariable::ProgramVariable(Value *value) {
 ProgramVariable::ProgramVariable(Value *value, int index) {
   this->value = value;
   this->rawName = dataflow::variable(value) + "." + std::to_string(index);
-  this->cleanedName = dataflow::variable(value) + "." + std::to_string(index);
+  this->cleanedName = this->rawName;
```

Comment:
Shouldn't this be a true copy? I'm concerned that this is just copying the pointer, and that the `erase` call on line 20 below will also affect `rawName`.

Reply:
This makes a true copy. See https://cplusplus.com/reference/string/string/operator=/. If I were to copy the pointer it would look like `this->cleanedName = &this->rawName" and this->cleanName would be a string pointer

Reply:
Great, sorry for my confusion. I'm slowly relearning C++ :)